app/Http/Controllers/Secretariat/SemesterEvaluationController.php

```diff
@@ -157,7 +157,7 @@ public function store(Request $request)
                         return back()->with('error', "A státusz megadása kötelező!")->with('section', $request->section);
                     }
                     $user->setStatusFor(Semester::next(), $request->next_status, $request->next_status_note);
-                    if ($request->has('resign_residency') && $user->isResident()) {
+                    if ($request->has('resign_residency') && $user->resides()) {
```

Comment:
resides() is a scope, which I think returns a query builder (test it in tinker). Check if these checks work as expected. In the tests it also seems fine, just double check it, I would not call scopes this way. I would rather make it a simple function